scripts/format_code.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A tool for manage source file formatting.
+
+Assembles a list of files via paths, directories, and/or git diffs and runs
+clang-format on them.  By default the tool will format the files in place.
+-noformat_file will run a check-only mode.
+
+Returns:
+   0: Either no files were found which need formatting, or if format_file was
+     enablled and the files were succesfully formatted.
+   1: Files which need formatting were found but weren't formatted
+     because -noformat_file was set.
+   2: The application wasn't configured properly and could not execute.
+"""
+import io
+import os
+import subprocess
+import sys
+
+from os import listdir
```

Comment:
Since os is already imported you dont have to specifically import functions from that module.
You can just use `os.listdir` (also easier to read and follow that listdir is from os module) and `os.path.isfile` etc.